dialogue-annotations-processor/src/main/java/com/palantir/dialogue/annotations/processor/generate/ServiceImplementationGenerator.java

```diff
@@ -99,9 +103,21 @@ private MethodSpec clientImpl(EndpointDefinition def) {
 
         def.arguments().forEach(arg -> generateParam(arg).ifPresent(methodBuilder::addCode));
 
-        methodBuilder.returns(def.returns());
+        methodBuilder.returns(def.returns().returnType());
+
+        boolean isAsync = def.returns().asyncInnerType().isPresent();
 
-        methodBuilder.addCode("throw new $T();", UnsupportedOperationException.class);
+        String executeCode = isAsync
+                ? "$L.clients().call($L, $L.build" + "(), $L);"
+                : "$L.clients().callBlocking($L, $L.build(), $L);";
+        CodeBlock execute = CodeBlock.of(
+                executeCode,
+                serviceDefinition.conjureRuntimeArgName(),
+                def.channelFieldName(),
+                REQUEST,
+                def.returns().deserializerFieldName());
+        methodBuilder.addCode(isAsync ? "return " : (def.returns().isVoid() ? "" : "return "));
+        methodBuilder.addCode(execute);
```

Comment:
Might be easier to read along these lines (untested, might be misremembering the api a bit):

```suggestion
        methodBuilder.addCode(def.returns().isVoid() && !isAsync ? "$L" : "return $L", execute);
```

Reply:
```methodBuilder.addCode(!def.returns().isVoid() || isAsync ? "return $L" : "$L", execute);```

works